Sources/NIOPosix/Thread.swift

```diff
@@ -42,41 +44,74 @@ protocol ThreadOps {
 /// All methods exposed are thread-safe.
 @usableFromInline
 final class NIOThread: Sendable {
-    internal typealias ThreadBoxValue = (body: (NIOThread) -> Void, name: String?)
+    internal typealias ThreadBoxValue = (body: (NIOThread) -> Void, name: String?, isDetached: Bool)
```

Comment:
Should we tweak this API to try to discourage creating detached threads outside of tests?